### DISCUSSION TITLE
[Usage] Is this compatible with...

### DISCUSSION THREAD

**Discussion by dhern023** [modin-project/modin#4266] [Q&A]
Pandas API extenders like pandas_flavor? Pandas_flavor allows you to attach new functions to dataframes, similar to the pipe function.

WSL? Do I need to have Ray or Dask on my machine to take full advantage of this?

**Comment by devin-petersohn** [comment]
Hi @dhern023!

>Pandas API extenders like pandas_flavor?

I am not sure this would work without some changes in the pandas_flavor project after glancing at the code in that repo.

>WSL?

Yes it is compatible, although I have noticed that Ray and Dask can both be 10x slower (sometimes more!) than on native windows. I use a windows machine as my daily development machine and use Modin on Ray or Dask every day. Do you have constraints that require WSL?

>Do I need to have Ray or Dask on my machine to take full advantage of this?

Not only those two. You can also use Omnisci (https://modin.readthedocs.io/en/stable/development/using_omnisci.html). Additionally, we have ongoing work to also support pure python multiprocessing, but the performance is not yet as good as either Ray or Dask. Efforts toward that are here: https://github.com/modin-project/unidist

**Reply by dhern023** [reply]
>  Ray and Dask can both be 10x slower 

I noticed this and ran into a lot of the common errors people have when trying to run it in WSL.

> Do you have constraints that require WSL?

Not really, I prefer to use it for speed. But I always aim for cross-compatibility, which isn't too hard with python.

> Not only those two.

So do I need to have Ray or Dask as installed packages or does modin handle that for me? I'm not familiar with what they are, or how to set them up.

**Reply by devin-petersohn** [reply]
If you take a look at the installation instructions: https://modin.readthedocs.io/en/stable/getting_started/installation.html, you can add an install target when you install with Modin which will add the engine you want (or both).

Modin will automatically manage this and start the engine in local mode for you. It happens the first time you create or load a dataframe, so you don't need to do any setup if you are on a single machine.
